test/examples_test.go

```diff
@@ -23,6 +25,37 @@ func TestCoinPricesFromCoingecko(t *testing.T) {
 		RunScript(filepath.Join(scriptDir, projectName+".txtar"))
 }
 
+func TestHelloWorldOnChain(t *testing.T) {
+	projectName := "hello_world_on_chain"
+	projectDir := filepath.Join(examplesDir, projectName)
+	npmDir := func(env *testscript.Env) (string, error) {
+		return env.WorkDir+"/.npm", nil
+	}
+	npxDir := func(env *testscript.Env) (string, error) {
+		return env.WorkDir+"/.npx", nil
+	}
```

Comment:
In the first iteration, I was setting these environment variables inside of the `NPMInstall()` setup function. Dave advised against this, however, as it could lead to test misconfigurations if devs did not realize `NPMInstall()` was also changing the environment.

Setting these environment variables is now explicit, but I had to use this "lazy" evaluation of the value because I need the testscript working dir which is not actually available until runtime.